Copy paket exe to cwd on init

Bomret opened the pull request:
As discussed in [Paket.Atom issue 14](https://github.com/fsprojects/Paket.Atom/issues/14) Paket should download the paket.bootstrapper into the .paket dir on init. I created a new module _Paket.Releases_ which handles downloading the latest versions of paket.exe, the bootstrapper and paket.targets.

forki commented:
/cc @theimowski could you please take a short look?

forki commented on src/Paket.Core/Releases.fs:
```diff
@@ -0,0 +1,61 @@
+        currentVersion < latest
+
+let downloadLatestVersionOf files destDir =
+    let releasesUrl = "https://api.github.com/repos/fsprojects/Paket/releases";
```
please move this into the Constants module

forki commented on src/Paket.Core/Releases.fs:
```diff
@@ -0,0 +1,61 @@
+
+let downloadLatestVersionOf files destDir =
+    let releasesUrl = "https://api.github.com/repos/fsprojects/Paket/releases";
+    use client = createWebClient("https://github.com",None)
```
Also Constants

forki commented on src/Paket.Core/Releases.fs:
```diff
@@ -0,0 +1,61 @@
+let downloadLatestBootstrapper environment =        
+    let exeDir = Path.Combine(environment.RootDirectory.FullName, ".paket")
+
+    downloadLatestVersionOf ["paket.bootstrapper.exe"] exeDir
```
I know this is all refactored from existing code, but please put all these constants into the Constants module

theimowski commented on src/Paket.Core/Releases.fs:
```diff
@@ -0,0 +1,61 @@
+
+    downloadLatestVersionOf ["paket.bootstrapper.exe"] exeDir
+
+let downloadLatestBootstrapperAndTargets environment =        
```
I'd rather inline this and above functions - they seem pretty straightforward

theimowski commented on src/Paket.Core/Releases.fs:
```diff
@@ -0,0 +1,61 @@
+        do! downloadFileSync url file.FullName client
+    }
+
+let private existsNotOrIsNewer (file:FileInfo) latest =
```
`doesNotExist` ?

theimowski commented:
apart from the comments, LGTM

Bomret commented:
I moved the URLs and other stuff into the Constants module and added comments on the functions. @theimowski what exactly did you mean by "inline these and above functions"? Remove the _downloadLatestBootstrapper*_ functions and only call _downloadLatestVersionOf_?

forki commented:
I think he meant, that you just remove the functions and use the code directly without a separate function

Bomret commented:
But this would be unnecessary duplication? The way it is now the _downloadLatestVersionOf_ function can be used to easily extend the functionality of the module later for other files.

forki commented:
Ok. Thanks a lot
On Jun 2, 2015 20:22, "Stefan Reichel" <notifications@github.com> wrote:

> But this would be unnecessary duplication? The way it is now the
> *downloadLatestVersionOf* function can be used to easily extend the
> functionality of the module later for other files.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/fsprojects/Paket/pull/853#issuecomment-108043204>.
>

Bomret commented:
And both other functions are used in the publicAPI module. Or do I still get it wrong?

forki commented:
thx. released in 1.10

theimowski commented:
Ok nvmd, I've not looked at the call sites for those functions - they just seemed extraordinary at first glance
